src/bidiMapper/domains/events/SubscriptionManager.ts

```diff
@@ -147,36 +161,51 @@ export class SubscriptionManager {
   }
 
   subscribe(
-    event: Session.SubscriptionRequestEvent,
-    contextId: CommonDataTypes.BrowsingContext | null,
+    event: ChromiumBidi.EventNames,
+    contextId: BrowsingContext.BrowsingContext | null,
     channel: string | null
   ): void {
     // All the subscriptions are handled on the top-level contexts.
     contextId = this.#browsingContextStorage.findTopLevelContextId(contextId);
 
-    if (event === BrowsingContext.AllEvents) {
-      Object.values(BrowsingContext.EventNames).map((specificEvent) =>
-        this.subscribe(specificEvent, contextId, channel)
-      );
-      return;
-    }
-    if (event === Log.AllEvents) {
-      Object.values(Log.EventNames).map((specificEvent) =>
-        this.subscribe(specificEvent, contextId, channel)
-      );
-      return;
-    }
-    if (event === Network.AllEvents) {
-      Object.values(Network.EventNames).map((specificEvent) =>
-        this.subscribe(specificEvent, contextId, channel)
-      );
-      return;
-    }
-    if (event === Script.AllEvents) {
-      Object.values(Script.EventNames).map((specificEvent) =>
-        this.subscribe(specificEvent, contextId, channel)
-      );
-      return;
+    switch (event) {
+      case ChromiumBidi.BrowsingContext.EventNames.AllBrowsingContextEvent:
+        Object.values(ChromiumBidi.BrowsingContext.EventNames)
+          .filter(
+            (name) =>
+              name !==
+              ChromiumBidi.BrowsingContext.EventNames.AllBrowsingContextEvent
+          )
+          .map((specificEvent) =>
+            this.subscribe(specificEvent, contextId, channel)
+          );
+        break;
+      case ChromiumBidi.Log.EventNames.AllLogEvent:
+        Object.values(ChromiumBidi.Log.EventNames)
+          .filter((name) => name !== ChromiumBidi.Log.EventNames.AllLogEvent)
+          .map((specificEvent) =>
+            this.subscribe(specificEvent, contextId, channel)
+          );
+        break;
+      case ChromiumBidi.Network.EventNames.AllNetworkEvent:
+        Object.values(ChromiumBidi.Network.EventNames)
+          .filter(
+            (name) => name !== ChromiumBidi.Network.EventNames.AllNetworkEvent
+          )
+          .map((specificEvent) =>
+            this.subscribe(specificEvent, contextId, channel)
+          );
+        break;
+      case ChromiumBidi.Script.EventNames.AllScriptEvent:
+        Object.values(ChromiumBidi.Script.EventNames)
+          .filter(
+            (name) => name !== ChromiumBidi.Script.EventNames.AllScriptEvent
+          )
+          .map((specificEvent) =>
+            this.subscribe(specificEvent, contextId, channel)
+          );
+        break;
+      default:
```

Comment:
Forgotten `default:`

Reply:
I don't think this needs a `default:` (unless ESLint complains). This is specifically for the "all events". There are no other subscriptions to be handled after.

Suggestion: Remove the `default:`. If the linter complains, then replace the comment below with `// Intentionally left empty.` (or similar).